server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -951,12 +951,18 @@ private SearchSourceBuilder shallowCopy(QueryBuilder queryBuilder, QueryBuilder
         return rewrittenBuilder;
     }
 
+    public void parseXContent(XContentParser parser) throws IOException {
+        parseXContent(parser, false);
```

Comment:
Should this default not be the other way around (pass in true by default)? The only time I think we should allow trailing tokens is in the _msearch API which is a special case. It seems safer to me to check and error on trailing tokens in the default case and then have the _msearch API use the more specialised method below?